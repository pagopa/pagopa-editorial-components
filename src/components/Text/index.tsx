import { Typography } from '@mui/material';
import EContainer from '../EContainer';

export interface TextPros {
  title?: string;
  subtitle?: string;
  text?: string;
  textAlign?: 'center' | 'left';
}

const Text = (props: TextPros) => {
  const { title, subtitle, text, textAlign = 'left' } = props;
  return (
    <EContainer background="background.paper" py={16}>
      {title && (
        <Typography variant="h3" mb={2} textAlign={textAlign} width={'100%'}>
          {title}
        </Typography>
      )}
      {subtitle && (
        <Typography variant="h6" mb={2} textAlign={textAlign} width={'100%'}>
          {subtitle}
        </Typography>
      )}
      {text && (
        <Typography variant="body2" textAlign={textAlign} width={'100%'}>
          {text}
        </Typography>
      )}
    </EContainer>
  );
};

export default Text;
