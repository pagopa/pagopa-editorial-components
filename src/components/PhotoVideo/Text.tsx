import { type CommonProps } from 'types/components';
import { Typography } from '@mui/material';

export interface PhotoVideoTextProps extends CommonProps {
  title?: string;
  subtitle?: string;
}

const Text = ({ title, subtitle, theme = 'dark' }: PhotoVideoTextProps) => {
  const color = theme === 'dark' ? 'primary.contrastText' : 'text.primary';
  return (
    <>
      {title && (
        <Typography variant="h5" mb={4} color={color}>
          {title}
        </Typography>
      )}
      {subtitle && (
        <Typography paragraph sx={{ fontSize: '16px' }} mb={3} color={color}>
          {subtitle}
        </Typography>
      )}
    </>
  );
};

export default Text;
