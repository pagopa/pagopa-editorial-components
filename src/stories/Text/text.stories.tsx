import { type ComponentMeta, type ComponentStory } from '@storybook/react';
import Text from '../../components/Text';

export default {
  component: Text,
  title: 'Text/light',
} as ComponentMeta<typeof Text>;

const Template: ComponentStory<typeof Text> = (args) => <Text {...args} />;

export const Standard = Template.bind({});
Standard.args = {
  title: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
  subtitle:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
  text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
};

export const TextAlignCenter = Template.bind({});
TextAlignCenter.args = {
  textAlign: 'center',
  title: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit',
  subtitle:
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
  text: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.',
};
