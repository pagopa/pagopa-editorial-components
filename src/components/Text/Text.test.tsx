import Text from './index';
import '@testing-library/jest-dom';
/**
 * @jest-environment jsdom
 */

import renderer from 'react-test-renderer';

it('renders correctly standard ', () => {
  const tree = renderer
    .create(
      <Text
        subtitle="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
        text="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
        title="Lorem ipsum dolor sit amet, consectetur adipiscing elit"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly text align center ', () => {
  const tree = renderer
    .create(
      <Text
        subtitle="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
        text="Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua."
        textAlign="center"
        title="Lorem ipsum dolor sit amet, consectetur adipiscing elit"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
