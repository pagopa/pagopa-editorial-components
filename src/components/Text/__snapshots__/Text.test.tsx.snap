// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly standard  1`] = `
<section
  className="MuiBox-root css-1epl6ta"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-1p4tas-MuiContainer-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <h3
        className="MuiTypography-root MuiTypography-h3 css-1cd8ovr-MuiTypography-root"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit
      </h3>
      <h6
        className="MuiTypography-root MuiTypography-h6 css-u053ky-MuiTypography-root"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
      </h6>
      <p
        className="MuiTypography-root MuiTypography-body2 css-vwq3qa-MuiTypography-root"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
      </p>
    </div>
  </div>
</section>
`;

exports[`renders correctly text align center  1`] = `
<section
  className="MuiBox-root css-1epl6ta"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-1p4tas-MuiContainer-root"
  >
    <div
      className="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
    >
      <h3
        className="MuiTypography-root MuiTypography-h3 css-vunvij-MuiTypography-root"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit
      </h3>
      <h6
        className="MuiTypography-root MuiTypography-h6 css-a2anxz-MuiTypography-root"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
      </h6>
      <p
        className="MuiTypography-root MuiTypography-body2 css-z1n48v-MuiTypography-root"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
      </p>
    </div>
  </div>
</section>
`;
